### PR TITLE
Promote TokenRequest e2e test to Conformance

### PR DESCRIPTION
**Pull request by zshihang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
promote TokenRequest e2e test to Conformance to prevent regression

```release-note
NONE
```

### CODE REVIEW

**Comment by zshihang** [comment]
the test is running at https://testgrid.k8s.io/sig-auth-gce#gce&width=5&include-filter-by-regex=ServiceAccounts%20should%20mount%20projected%20service%20account%20token

**Comment by liggitt** [comment]
/triage accepted
/priority important-soon
/hold for soak

/lgtm
for sig-auth

File: test/e2e/auth/service_accounts.go [hunk 1]
```diff
@@ -424,7 +424,7 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 	  Testname: TokenRequestProjection should mount a projected volume with token using TokenRequest API.
 	  Description: Ensure that projected service account token is mounted.
 	*/
-	ginkgo.It("should mount projected service account token when requested", func() {
+	framework.ConformanceIt("should mount projected service account token", func() {
```
Comment (oomichi) [inline test/e2e/auth/service_accounts.go:427]: This test has been added 8 days ago.
The conformance test should be running stably at least 2 weeks as
`it is stable and runs consistently (e.g., no flakes), and has been running for at least two weeks`
from https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements
We need to wait for more days to pass the above requirement.

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

**Comment by liggitt** [comment]
/hold cancel

test is 100% green over the last 15 days

**Pull request merged**